Source/Dafny/DafnyConsolePrinter.cs

```diff
@@ -57,4 +58,32 @@ class DafnyConsolePrinter : ConsolePrinter {
       ReportBplError(nt.Inner, "Related location", false, tw);
     }
   }
+
+  public override void ReportEndVerifyImplementation(Implementation implementation, Boogie.VerificationResult result) {
+    verificationResults.Add((implementation, result));
+  }
```

Comment:
We can use this hook to raise test logger events now instead of reading the `/xml` file now, right?

(in a separate PR of course :)

Reply:
Yep, I've been starting to think that.